### PR TITLE
Add instructions on how to add enumerations to SAL XML

### PR DESCRIPTION
**Pull request by couger01**

### CODE REVIEW

File: doc/sal_constraints_and_recommendations.rst [hunk 1]
```diff
@@ -452,6 +453,14 @@ e.g. for ATTCS in C++:
...
     ATTCS_someEvent_SpecificSetA (datum specific)
 
+Define it using the following format.
+
+1. In ``CSCName_Events.xml`` create an ``Enumeration`` tag just before the first ``SAlEvent`` tag in the file.
+2. Fill out the Enumeration with comma delimited values using the following naming format ``itemName_value``.
+3. Repeat for as many enumerations as you require.
+
+.. warning:: Some enumerations are defined inside of the topic themselves. This is a legacy method and no longer recommended or necessary.
+
```
Comment (pkubanek) [inline doc/sal_constraints_and_recommendations.rst:463]: Shouldn't you mention you shall (optionally) add enumerations values as constant to ts_idl/python/lsst/ts/idl/enums/<CSCName.py? At least before that would be auto-generated?

File: doc/sal_constraints_and_recommendations.rst [hunk 2]
```diff
@@ -452,6 +453,15 @@ e.g. for ATTCS in C++:
     ATTCS_shared_SimpleSetA (declared globally)
     ATTCS_someEvent_SpecificSetA (datum specific)
 
+Define it using the following format.
+
+1. In ``CSCName_Events.xml`` create an ``Enumeration`` tag just before the first ``SAlEvent`` tag in the file.
+2. Fill out the Enumeration with comma delimited values using the following naming format ``itemName_value``.
+3. Repeat for as many enumerations as you require.
+4. Optionally, add the enumerations to ``ts_idl`` by creating a ``lsst.ts.idl.enums.{csc_name}`` module.
```
Comment (tribeiro) [inline doc/sal_constraints_and_recommendations.rst:461]: Can you remove the "Optionally"?

Comment (couger01) [reply doc/sal_constraints_and_recommendations.rst]: Done.

Comment (pkubanek) [reply doc/sal_constraints_and_recommendations.rst]: Well they might be cases (C++ only) when this is not needed. And I believe we target to autogenerate those (which shall not be so difficult..it's just XML parsing, I can add this feature if requested to make_idl_files.py)

Comment (couger01) [reply doc/sal_constraints_and_recommendations.rst]: But it is helpful for ts_salobj remotes.

**Review by pkubanek — APPROVED** [review]
Looks good. Thanks for adding ts_idle enums to description.

**Pull request merged**